classes/checkout/CheckoutProcess.php

```diff
@@ -100,6 +100,20 @@ public function addStep(CheckoutStepInterface $step)
         return $this;
     }
 
+    public function removeStepByIdentifier(string $stepIdentifier): self
```

Comment:
Hi  @eternoendless,

this is a super good idea ! Let me improve that :+1: 

Cheers

Reply:
ping @eternoendless, I'd live to reuse Abstract Collection to avoid passing an array by reference (and honestly, this is error prone: I always forgot to add this extra "&" when manipulating arrays in the hooks).

WDYT ?